test/indentation-test.el

```diff
@@ -0,0 +1,71 @@
+(require 'ert)
+
+(load-file "groovy-mode.el")
+
+(ert-deftest test-function-indentation-using-groovy-specific-function ()
+  (with-temp-buffer
+    (let ((text "def f() {
+def a = 1
+}
+"))
+      (insert text)
+      (goto-char (point-min))
+      (groovy-indent-line)
+      (next-line)
+      (groovy-indent-line)
+      (next-line)
+      (groovy-indent-line)
+      (should (equal (buffer-string) "def f() {
+    def a = 1
+}
+")))))
+
+(ert-deftest test-function-indentation-using-generic-function ()
+  (with-temp-buffer
+    (let ((text "def f() {
+def a = 1
+}
+"))
+      (groovy-mode)
+      (insert text)
+      (goto-char (point-min))
+      (indent-line-function)
```

Comment:
For what it's worth, this doesn't work because `indent-line-function` is a variable that points to a function. Emacs lisp is a lisp-2: variables and functions are in different namespaces. You'd need `(funcall indent-line-function)` instead.